warehouse/packaging/tasks.py

```diff
@@ -211,130 +208,10 @@ def reconcile_file_storages(request):
                 file.cached = True
 
 
-@tasks.task(ignore_result=True, acks_late=True)
-def compute_2fa_mandate(request):
-    # Get our own production dependencies
-    our_dependencies = set(
-        pip_api.parse_requirements("./requirements/main.txt")
-        | pip_api.parse_requirements("./requirements/deploy.txt")
-    )
-
-    bq = request.find_service(name="gcloud.bigquery")
-
-    # Get the top N projects in the last 6 months
-    query = bq.query(
-        """ SELECT
-              COUNT(*) AS num_downloads,
-              file.project as project_name
-            FROM
-              {table}
-            WHERE
-              DATE(timestamp) BETWEEN DATE_TRUNC(
-                DATE_SUB(CURRENT_DATE(), INTERVAL 6 MONTH), MONTH
-              )
-              AND CURRENT_DATE()
-            GROUP BY
-              file.project
-            ORDER BY
-              num_downloads DESC
-            LIMIT
-              {cohort_size}
-        """.format(
-            table=request.registry.settings["warehouse.downloads_table"],
-            cohort_size=request.registry.settings[
-                "warehouse.two_factor_mandate.cohort_size"
-            ],
-        )
-    )
-    top_projects = {row.get("project_name") for row in query.result()}
-
-    project_names = {canonicalize_name(n) for n in our_dependencies | top_projects}
-
-    # Get the projects that were not previously in the mandate
-    new_projects = request.db.query(Project).filter(
-        Project.normalized_name.in_(project_names), Project.pypi_mandates_2fa.is_(False)
-    )
-
-    # Add them to the mandate
-    new_projects.update({Project.pypi_mandates_2fa: True})
-
-
 @tasks.task(ignore_result=True, acks_late=True)
 def compute_2fa_metrics(request):
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought (non-blocking):** ‏This is likely to break some of the metrics on our publicly-shared 2FA Dashboard. Not a problem per se, but something we should be aware of.

Reply:
I've already updated the dashboard accordingly.